File: core/src/main/java/com/mstream/rulesengine/core/ValidationResult.java
package com.mstream.rulesengine.core;

import java.util.Map;
import java.util.Set;

public interface ValidationResult {

    Set<String> getFacts();

    Set<String> getActions();

    Map<String, String> getErrors();

    Map<String, String> getWarnings();
}

File: core/src/main/clojure/com/mstream/rules_engine/java/validation_result.clj
(ns com.mstream.rules-engine.java.validation-result)


(defrecord ValidationResult
  [errors
   warnings
   actions
   facts]
  com.mstream.rulesengine.core.ValidationResult
  (getErrors [_]
    errors)
  (getWarnings [_]
    warnings)
  (getActions [_]
    actions)
  (getFacts [_]
    facts)
  (toString [_]
    (str {:errors   errors
          :warnings warnings
          :actions  actions
          :facts    facts})))
